File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/Handlers/UpdateOrderStatusCommandHandler.cs
using MediatR;
using MediatorDesignPattern.Example2.Commands;
using MediatorDesignPattern.Example2.Models;
using MediatorDesignPattern.Example2.Repositories;

namespace MediatorDesignPattern.Example2.Handlers;

/// <summary>
/// Handler for updating order status
/// </summary>
public class UpdateOrderStatusCommandHandler : IRequestHandler<UpdateOrderStatusCommand, Order?>
{
    private readonly IOrderRepository _orderRepository;

    public UpdateOrderStatusCommandHandler(IOrderRepository orderRepository)
    {
        _orderRepository = orderRepository;
    }

    public Task<Order?> Handle(UpdateOrderStatusCommand request, CancellationToken cancellationToken)
    {
        var order = _orderRepository.GetById(request.OrderId);
        
        if (order == null)
        {
            Console.WriteLine($"[HANDLER] Order not found: {request.OrderId}");
            return Task.FromResult<Order?>(null);
        }

        order.Status = request.NewStatus;
        order.UpdatedAt = DateTime.UtcNow;
        
        _orderRepository.Update(order);

        Console.WriteLine($"[HANDLER] Order status updated: {order.Id} -> {order.Status}");
        
        return Task.FromResult<Order?>(order);
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/src/Handlers/UpdateOrderStatusCommandHandler.ts
import { requestHandler, RequestHandler } from 'mediatr-ts';
import { UpdateOrderStatusCommand } from '../Commands/UpdateOrderStatusCommand';
import { Order } from '../Models/Order';
import { orderRepository } from './CreateOrderCommandHandler';

@requestHandler(UpdateOrderStatusCommand)
export class UpdateOrderStatusCommandHandler implements RequestHandler<UpdateOrderStatusCommand, Order | undefined> {
  async handle(request: UpdateOrderStatusCommand): Promise<Order | undefined> {
    const order = orderRepository.getById(request.orderId);

    if (!order) {
      console.log(`[HANDLER] Order not found: ${request.orderId}`);
      return undefined;
    }

    order.status = request.newStatus;
    order.updatedAt = new Date();
    orderRepository.update(order);

    console.log(`[HANDLER] Order status updated: ${order.id} -> ${order.status}`);
    return order;
  }
}
